src/NuGet.Protocol.Core.v3/DataClient/DataClient.cs

```diff
@@ -124,6 +124,11 @@ public JObject GetJObject(Uri address)
         public async Task<JObject> GetJObjectAsync(Uri address, CancellationToken token)
         {
             var response = await GetAsync(address, token);
+            if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
+            {
+                throw new UnauthorizedAccessException();
+            }
+
```

Comment:
The first change is that this method throws UnauthorizedAccessException() on authentication failure.